### DISCUSSION TITLE
OnInserted status effect triggers even when you fail to insert

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15283] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

```xml
<Item name="Experimental Rifle" identifier="ex_rifle" category="Weapon" cargocontaineridentifier="metalcrate" tags="mediumitem,weapon,gun,gunsmith,provocativetohumanai,mountableweapon" Scale="0.5" impactsoundtag="impact_metal_light">
  <Sprite texture="Content/Items/Weapons/weapons_new.png" sourcerect="254,2,200,55" depth="0.55" origin="0.5,0.25" />
  <Body width="190" height="50" density="25" />
  <Holdable slots="Any,RightHand+LeftHand" controlpose="true" holdpos="40,-10" aimpos="55,5" handle1="-25,-16" handle2="26,0" holdangle="-40" msg="ItemMsgPickUpSelect" />
  <ItemContainer capacity="1" maxstacksize="1" hideitems="false" containedstateindicatorstyle="bullet" containedstateindicatorslot="1" containedspritedepth="0.56">
    <Containable items="grenade" hide="false" itempos="50,0">
      <StatusEffect type="OnInserted" target="This">
        <Sound file="Content/Items/Weapons/WEAPON_rifleGrenadeShot1.ogg" range="1000" />
      </StatusEffect>
    </Containable>
    <SlotIcon slotindex="0" texture="Content/UI/StatusMonitorUI.png" sourcerect="448,448,64,64" origin="0.5,0.5" />
    <SlotIcon slotindex="1" texture="Content/UI/StatusMonitorUI.png" sourcerect="256,448,64,64" origin="0.5,0.5" />
    <SubContainer capacity="1" maxstacksize="2">
      <Containable items="rifleammo" hide="true">
        <StatusEffect type="OnInserted" target="This" condition="-5" disabledeltatime="true">
          <Sound file="Content/Items/Weapons/WEAPON_rifleReload50-_1.ogg" range="500" />
        </StatusEffect>
      </Containable>
    </SubContainer>
  </ItemContainer>
</Item>
```

### Reproduction steps

1. test item "Experimental Rifle"
2. insert 2 rifle bullets inside -> weapon condition will decrease 10% expected behavior
3. try to insert smg magazine into bullet slot -> weapon condition will decrease 15%

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/ac937622-ec55-46ca-bde6-3f67634c6e20

**Comment by 4-S-H** [comment]
Happened to me as well.

**Comment by Regalis11** [comment]
~~First of all, sorry for taking so long to investigate this! To me it seems like this issue has been fixed in the latest version (most likely by the fixes to the `OnInserted` effects in v1.9.6.0, the Summer 2025 Update). Closing the ticket.~~

https://github.com/user-attachments/assets/d6c55097-ff85-4467-b4f3-940e6c424ce9

~~If you're still encountering it though, please let me know!~~

EDIT: ah, I was attempting to reproduce this incorrectly, it seems it only happens if the rifle bullets are still in the gun and I'm try to swap them with the magazine.

**Comment by Regalis11** [comment]
Addressed in our private repo now, fix coming up in the next unstable build and the upcoming Spring Update.
